Workiva/ixbrl-viewer
# XT-2948 - Create puppeteer JS config/smoke test for ixbrl-viewer

#### Reason for change
Additional browser based testing will help ensure overall product quality/compatibility.

#### Description of change
* Created a puppeteer config that runs in a github workflow against Window/MacOS/Ubuntu with multiple nodejs versions.  This workflow generates the viewer js, installs Arelle, generates a test project, then runs the supplied test against the generated file.   Should the test fail, it will store off a video of the run, as well as the generated viewers.
* Created a workflow/documentation for running the tests locally.
* There are compatibility issues with `puppeteer` and the version of `jest` defined in the repo, so I went ahead and bumped `jest` up to latest.

#### Steps to Test
* Github actions should include the new `Run Puppeteer Test` config in the various configurations
* I added instructions to the readme on how to run it locally, go ahead and give them a try to make sure they work.

**review**:
@Workiva/xt
@paulwarren-wk
